nexus_constructor/main_window.py

```diff
@@ -169,6 +171,32 @@ def show_edit_component_dialog(self):
         ].internalPointer()
         self.show_add_component_window(selected_component)
 
+    def _setupFileWriterCtrl(self, main_window):
+        try:
+            import kafka  # noqa: F401
+
+            self.control_file_writer_action = QtWidgets.QAction(main_window)
```

Comment:
Can this be moved to main_window.py in `ui/`? i also don't think this is getting added to the main layout, so at the moment won't be visible.